src/main/java/pl/javastart/task/StationUtils.java

```diff
@@ -0,0 +1,127 @@
+package pl.javastart.task;
+
+import java.io.*;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Scanner;
+
+public class StationUtils {
+    private Queue<Vehicle> vehicles = new LinkedList<>();
+
+    public void run(String fileName) throws IOException {
+        if (!fileName.isEmpty()) {
+            readQueueFromFile(fileName);
+        }
+        boolean canContinue = true;
+        while (canContinue) {
+            printUserChoice();
+            int userChoice = readUserChoice();
+            if (userChoice == 0) {
+                canContinue = fileWritingAction(fileName);
+            } else if (userChoice == 1) {
+                vehicles.offer(createVehicle());
+            } else if (userChoice == 2) {
+                inspectionIfQueueNotEmpty(fileName);
+            } else {
+                System.out.println("Coś poszło nie tak");
+            }
+        }
+    }
+
+    public void performInspection(String fileName) throws IOException {
+        if (!vehicles.isEmpty()) {
+            Vehicle vehicleToInspection = vehicles.poll();
+            System.out.println("Dokonano przeglądu pojazdu o numerze VIN: " + vehicleToInspection.getVin());
+        } else {
+            clearFile(fileName);
+        }
```

Comment:
Czyszczeniem pliku zajmujesz się w metodzie `fileWritingAction` wywoływanej przy zamykaniu programu, nie widzę potrzeby, żeby robić to także tutaj dla pustej kolejki